docs/installation/server-config/security.rst

```diff
@@ -0,0 +1,163 @@
+.. _ksql-security:
+
+Securing KSQL
+-------------
+
+To connect to a secured Kafka cluster, Kafka client applications must provide their security credentials. In the same way,
+KSQL is configured so that the KSQL servers are authenticated and authorized, and data communication is encrypted when
+communicating with the Kafka cluster. KSQL can be configured for:
+
+- :ref:`SSL for encryption <kafka_ssl_encryption>`
+- :ref:`SASL for authentication <kafka_sasl_auth>`
+- :ref:`HTTPS for Confluent Schema Registry <schema_registry_http_https>`
+
+-------------------------------------------------
+Using KSQL with an Apache Kafka Secured with ACLs
+-------------------------------------------------
+
+You can use KSQL with Apache Kafka clusters that are secured with ACLs. The behavior depends on whether the cluster is
+interactive or non-interactive.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Interactive KSQL clusters require that the KSQL user has open access to create, read, write and delete topics and
+use any consumer group.
+
+The required ACLs are:
+
+- *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+- *CREATE* permission on the *CLUSTER*.
+- DESCRIBE*, *READ*, *WRITE* and *DELETE* permissions on the *<any>* *TOPIC*.
+- *DESCRIBE* and *READ* permissions  on the *<any>* *GROUP*.
+
+It is still possible to restrict the KSQL user from accessing specific resources using *DENY* ACLs. For example, you can add a
+*DENY* ACL to stop KSQL queries from accessing a topic that contains sensitive data.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Non-Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Non-interactive KSQL clusters will run with much more restrictive ACLs, though it currently requires a little effort to
+work out what ACLs are required. This will be improved in future KSQL releases.
+
+Standard ACLs
+    The KSQL user always requires:
+
+    - *DESCRIBE_CONFIGS* permission on the *CLUSTER*.
+    - *DESCRIBE* permission on the *__consumer_offsets* topic.
+
+    If you want KSQL to create internal and sink topics then the KSQL user should also be granted:
+
+    - *CREATE* permission on the *CLUSTER*.
+
+Source topics
+    KSQL users require *DESCRIBE* and *READ* permissions for each source and input topic. The topic should already exist
+    when KSQL is started.
+
+Sink topics
+    KSQL users require *DESCRIBE* and *WRITE* permissions ror each sink and output topic. If the topic does not already
+    exist, the user also requires *CREATE* permissions on the *CLUSTER*.
+
+Change-log and repartition topics
```

Comment:
These are internal topics as far as KSQL is concerned, so why are they listed here again, and separately from "internal" topics?

Reply:
@big-andy-coates ^

Reply:
There is no 'internal' topics section... :D

I'm assuming your referring to the section under `Standard ACLs` that talks about CREATE permissions?  Maybe we should drop that part and just cover it in `Output topics` and `internal topics`?

How about adding here something like:

> KSQL internally makes use of _repartition_ topics and _changelog_ topics for some operations. The KSQL user will require a minimum of *DESCRIBE*, *READ* and *WRITE* permissions for each changelog and repartition *TOPIC*.

>By default, KSQL will attempt to create any _repartition_ or _changelog_ topics that do not exist. To allow this, the KSQL user will require CREATE permissions on the CLUSTER. Alternatively, topics can be created manual before running KSQL. To determine the list of topics and their required configuration, (partition count, replication factor, retention policy, etc), it is recommended that you run KSQL on open cluster first.

> KSQL requires _repartition_ topics when using either `PARTITION BY`, or using `GROUP BY` on none-key values. KSQL requires  _changelog_ topics to back any `CREATE TABLE x AS` statements.

> All changelog and repartition topics are prefixed with  ``_confluent-ksql-<value of ksql.service.id property>_query_<query id>_`` where ``ksql.service.id`` defaults to ``ksql_``.

Side note: with a recent change that @hjafarpour put in, I think the default prefix for internal topics is now `_confluent-ksql-ksql_query`... note the double `ksql`

Reply:
@big-andy-coates i have replaced the existing content with this ^